sdk/appconfiguration/azure-appconfiguration-provider/azure/appconfiguration/provider/aio/_azureappconfigurationproviderasync.py

```diff
@@ -303,25 +325,42 @@ async def refresh(self, **kwargs) -> None:
                 if not self._refresh_timer.needs_refresh():
                     logging.debug("Refresh called but refresh interval not elapsed.")
                     return
+                need_refresh = False
+                updated_sentinel_keys = dict(self._refresh_on)
+                headers = _get_headers("Watch")
                 for (key, label), etag in self._refresh_on.items():
-                    updated_sentinel = await self._client.get_configuration_setting(
-                        key=key,
-                        label=label,
-                        etag=etag,
-                        match_condition=MatchConditions.IfModified,
-                        headers=_get_headers("Watch"),
-                        **kwargs
-                    )
-                    if updated_sentinel is not None:
-                        logging.debug(
-                            "Refresh all triggered by key: %s label %s.",
-                            key,
-                            label,
+                    try:
+                        updated_sentinel = await self._client.get_configuration_setting(
+                            key=key,
+                            label=label,
+                            etag=etag,
+                            match_condition=MatchConditions.IfModified,
+                            headers=headers,
+                            **kwargs
                         )
-                        await self._load_all(headers=_get_headers("Watch", **kwargs))
-                        self._refresh_on[(key, label)] = updated_sentinel.etag
-                        self._refresh_timer.reset()
-                        return
+                        if updated_sentinel is not None:
+                            logging.debug("Refresh all triggered by key: %s label %s.", key, label)
+                            need_refresh = True
+
+                            updated_sentinel_keys[(key, label)] = updated_sentinel.etag
+                    except HttpResponseError as e:
+                        if e.status_code == 404:
+                            if etag is not None:
+                                # If the sentinel is not found, it means the key/label was deleted, so we should refresh
+                                logging.debug("Refresh all triggered by key: %s label %s.", key, label)
+                                need_refresh = True
+                                updated_sentinel_keys[(key, label)] = None
+                        else:
+                            raise e
+                # Need to only update once, no matter how many sentinels are updated
+                if need_refresh:
+                    await self._load_all(**kwargs)
+                    self._refresh_on = updated_sentinel_keys
```

Comment:
`_load_all` updates ` self._refresh_on` in the sync version, so we dont need to update it in `refresh` method. Can we keep the same logic in async version?